python/tvm/relay/op/transform.py

```diff
@@ -249,6 +249,49 @@ def full_like(data, fill_value):
     return _make.full_like(data, fill_value)
 
 
+def arange(stop, start=None, step=1, dtype="float32"):
+    """Return evenly spaced values within a given interval.
+
+    Warning: Undefined behavior when dtype is incompatible with start/stop/step.
+    It could lead to different results compared to numpy, MXNet, pytorch, etc.
+
+    Parameters
+    ----------
+    stop : tvm.Expr
+        Stop of interval. The interval does not include this value.
+
+    start : tvm.Expr, optional
+        Start of interval. The interval includes this value. The default start
+        value is 0.
+
+    step : tvm.Expr, optional
+        Spacing between values. The default step size is 1.
+
+    dtype : str, optional
+        The target data type.
+
+    Returns
+    -------
+    result : relay.Expr
+        The resulting tensor.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        relay.arange(5) = [0, 1, 2, 3, 4]
+        relay.arange(1, 5) = [1, 2, 3, 4]
+        relay.arange(1, 5, 1.5) = [1, 2.5, 4]
+    """
```

Comment:
should we consider cases like `start > stop` and `step <= 0`, here? I think we probably need to at least warning or raise exceptions for `step == 0`

Reply:
Added the sanity check in the new commit